Asignment7/baseball_stats.sql

```diff
@@ -0,0 +1,485 @@
+    SELECT t1.team_id,
+           t1.game_id,
+           t1.local_date,
+           IF (SUM(t2.p_AtBat)=0,0,SUM(t2.p_Hit)/(SUM(t2.p_AtBat))) AS t_rolling_pitchingaverage
+        FROM s_team_pitching_stats AS t1 JOIN s_team_pitching_stats AS t2
+        ON t1.team_id= t2.team_id
+        WHERE t2.local_date >= DATE_ADD(t1.local_date, INTERVAL -200 DAY)
+        AND t2.local_date< t1.local_date
+     GROUP BY t1.team_id, t1.local_date,t1.game_id;
+
+-- Team rolling home runs scored
+
+DROP TABLE IF EXISTS s_team_home_runs_scored;
+CREATE TABLE s_team_home_runs_scored
+     (INDEX Game_ID_IND(game_id),INDEX Team_ID_IND(team_id), INDEX HOME_RUNS(home_runs))
+    SELECT t1.team_id,
+           t1.game_id,
+           t1.local_date,
+           SUM(t2.b_Home_Run) AS home_runs
+        FROM s_team_batting_stats AS t1 JOIN s_team_batting_stats AS t2
+        ON t1.team_id= t2.team_id
+        WHERE t2.local_date >= DATE_ADD(t1.local_date, INTERVAL -200 DAY)
+        AND t2.local_date< t1.local_date
+     GROUP BY t1.team_id, t1.local_date,t1.game_id;
+
+-- calculating BABIP  for teams
+
+DROP TABLE IF EXISTS s_team_rolling_BABIP;
+CREATE TABLE s_team_rolling_BABIP
+    (INDEX GAME_ID_IND(game_id),INDEX TEAM_ID_IND(team_id), INDEX b_team_BABIP(b_team_BABIP))
+    SELECT t1.team_id,
+           t1.game_id,
+           t1.local_date,
+           IF((sum(t2.b_AtBat) - SUM(t2.b_StrikeOut) - SUM(t2.b_Home_Run) + SUM(t2.b_Sac_Fly)) = 0, 0,
+                    (SUM(t2.b_Hit) - SUM(t2.b_Home_Run))/(sum(t2.b_AtBat) - SUM(t2.b_StrikeOut) - SUM(t2.b_Home_Run) + SUM(t2.b_Sac_Fly))) as b_team_BABIP
+        FROM s_team_batting_stats AS t1 JOIN s_team_batting_stats AS t2
+        ON t1.team_id= t2.team_id
+        WHERE t2.local_date >= DATE_ADD(t1.local_date, INTERVAL -200 DAY)
+        AND t2.local_date< t1.local_date
+     GROUP BY t1.team_id, t1.local_date,t1.game_id;
+SELECT * FROM team_rollingBABIP LIMIT 0,20;
+SELECT count((game_id))FROM team_rollingBABIP;
+
+-- calculating rolling BAIP for teams
+
+DROP TABLE IF EXISTS s_team_rolling_BAIP;
+CREATE TABLE s_team_rolling_BAIP
+    (INDEX GAME_IND(game_id),INDEX TEAM_IND(team_id), INDEX p_team_BAIP(p_team_BAIP))
+        SELECT t1.team_id,
+               t1.game_id,
+               t1.local_date,
+               IF((sum(t2.p_AtBat) - SUM(t2.p_StrikeOut) - SUM(t2.p_Home_Run) + SUM(t2.p_Sac_Fly)) = 0, 0,
+                            (SUM(t2.p_Hit) - SUM(t2.p_Home_Run))/(sum(t2.p_AtBat) - SUM(t2.p_StrikeOut) - SUM(t2.p_Home_Run) + SUM(t2.p_Sac_Fly))) as p_team_BAIP
+            FROM s_team_pitching_stats AS t1 JOIN s_team_pitching_stats AS t2
+            ON t1.team_id= t2.team_id
+            WHERE t2.local_date >= DATE_ADD(t1.local_date, INTERVAL -200 DAY)
+            AND t2.local_date< t1.local_date
+         GROUP BY t1.team_id, t1.local_date,t1.game_id;
+
+-- Calculating the rolling BB/k(walk to strike out ratio for batter) for teams
```

Comment:
I can see you have created multiple tables for creating individual features for the batting and pitching teams. You can calculate all features in one single temp table together for the batting and pitching teams respectively.